.github/workflows/main.yml

```diff
@@ -192,7 +190,8 @@ jobs:
         mkdir mdbook
         curl -Lf https://github.com/rust-lang/mdBook/releases/download/v0.4.27/mdbook-v0.4.27-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
         echo `pwd`/mdbook >> $GITHUB_PATH
-    - run: cargo doc --document-private-items --no-deps
+    # TODO: should check all workspace members
+    - run: cargo doc -p cargo --document-private-items --no-deps
```

Comment:
I think we should check the entire workspace.

Reply:
Definitely yes! However, I may leave it for future PRs. I want to make this PR focus on making itself a workspace. It is already something hard to do it correctly 🤯.